optaplanner-core/src/main/java/org/optaplanner/core/api/score/constraint/ConstraintMatch.java

```diff
@@ -28,6 +27,16 @@
 /**
  * Retrievable from {@link ConstraintMatchTotal#getConstraintMatchSet()}
  * and {@link Indictment#getConstraintMatchSet()}.
+ *
+ * <p>
+ * Note: this class has a {@link #compareTo(ConstraintMatch)} method which is inconsistent with equals.
```

Comment:
(soft) remove "Note", as all javadocs are a notes

Also link "equals" and mention hashcode (also linked).